util/containers/lru.go

```diff
@@ -3,30 +3,61 @@
 
 package containers
 
-import "github.com/golang/groupcache/lru"
+import (
+	"github.com/hashicorp/golang-lru/simplelru"
```

Comment:
there is a v2 that supports generics

https://pkg.go.dev/github.com/hashicorp/golang-lru/v2#Cache